cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -43,7 +44,7 @@ pub const FLOX_CATALOG_DUMP_DATA_VAR: &str = "_FLOX_CATALOG_DUMP_RESPONSE_FILE";
 static GENERATED_DATA: Lazy<PathBuf> =
     Lazy::new(|| PathBuf::from(std::env::var("GENERATED_DATA").unwrap()));
 
-const RESPONSE_PAGE_SIZE: NonZeroU32 = unsafe { NonZeroU32::new_unchecked(10) };
+const RESPONSE_PAGE_SIZE: NonZeroU32 = unsafe { NonZeroU32::new_unchecked(100000) };
```

Comment:
A default of 100k makes me feel _mildly uncomfortable_ because it's hard to wind back from if we need to protect the server from excessive resource consumption in the future and it's optimising for scenarios where the user experience is already poor or undefined (e.g. what is somebody doing with 42k results from `flox search th --all` dumped to their terminal?). This feels vaguely similar to the "bonus features" that we discussed in the catalog retro.

I'd vote for starting at 1k if that helps the known cases (e.g. `flox show bash` and `flox search java --all`) and then adjust again, or even remove pagination, if we find that it's desirable later.

Reply:
My thinking was that if we want a limit, we ought to implement a max page size on the server side rather than rely on the client being cooperative. The desired behavior here is basically "give me as much as you're willing to, because I'm going to read the whole result set anyway". The benefit of leaving the paging code in place is that the client can already gracefully handle the case where it receives a smaller page size than it requested, as the pagination controls are in the response.

Reply:
> The benefit of leaving the paging code in place is that the client can already gracefully handle the case where it receives a smaller page size than it requested, as the pagination controls are in the response.

I think right now we treat a response with fewer results than page size as being the last page

Reply:
Agree about implementing the max page on the server side and that we should keep the pagination code at either end now that we have it.

However if we restrict the server-side max today and it's less than the client's default, which it definitely would be at 100k, then the client will stop short of the first page. We only expose the number of items in the response rather than current/next/etc.

Reply:
Ahh you're right, I take back everything! In that case, this change definitely _would_ paint us into an unpleasant corner. I will revert it back to 1k, and we should separately have a conversation about `search --all`.